lib/compat/wordpress-6.7/class-gutenberg-rest-posts-controller-6-7.php

```diff
@@ -0,0 +1,937 @@
+				'before' => $request['before'],
+				'column' => 'post_date',
+			);
+		}
+
+		if ( isset( $registered['modified_before'], $request['modified_before'] ) ) {
+			$args['date_query'][] = array(
+				'before' => $request['modified_before'],
+				'column' => 'post_modified',
+			);
+		}
+
+		if ( isset( $registered['after'], $request['after'] ) ) {
+			$args['date_query'][] = array(
+				'after'  => $request['after'],
+				'column' => 'post_date',
+			);
+		}
+
+		if ( isset( $registered['modified_after'], $request['modified_after'] ) ) {
+			$args['date_query'][] = array(
+				'after'  => $request['modified_after'],
+				'column' => 'post_modified',
+			);
+		}
+
+		// Ensure our per_page parameter overrides any provided posts_per_page filter.
+		if ( isset( $registered['per_page'] ) ) {
+			$args['posts_per_page'] = $request['per_page'];
+		}
+
+		if ( isset( $registered['sticky'], $request['sticky'] ) ) {
+			$sticky_posts = get_option( 'sticky_posts', array() );
+			if ( ! is_array( $sticky_posts ) ) {
+				$sticky_posts = array();
+			}
+			if ( $request['sticky'] ) {
+				/*
+				 * As post__in will be used to only get sticky posts,
+				 * we have to support the case where post__in was already
+				 * specified.
+				 */
+				$args['post__in'] = $args['post__in'] ? array_intersect( $sticky_posts, $args['post__in'] ) : $sticky_posts;
+
+				/*
+				 * If we intersected, but there are no post IDs in common,
+				 * WP_Query won't return "no posts" for post__in = array()
+				 * so we have to fake it a bit.
+				 */
+				if ( ! $args['post__in'] ) {
+					$args['post__in'] = array( 0 );
+				}
+			} elseif ( $sticky_posts ) {
+				/*
+				 * As post___not_in will be used to only get posts that
+				 * are not sticky, we have to support the case where post__not_in
+				 * was already specified.
+				 */
+				$args['post__not_in'] = array_merge( $args['post__not_in'], $sticky_posts );
+			}
+		}
+
+		$args = $this->prepare_tax_query( $args, $request );
+
+		if ( isset( $request['format'] ) && ! empty( $request['format'] ) ) {
+			// If format is not an array, convert it to an array so that the
+			// required prefix can be added to all items.
+			$formats = is_array( $request['format'] ) ? $request['format'] : array( $request['format'] );
+
+			$tax_query = array( 'relation' => 'OR' );
+
+			// The default post format, 'standard', is not saved in the database.
+			// If 'standard' is part of the request, the query needs to exclude the formats.
+			if ( in_array( 'standard', $formats, true ) ) {
+				$tax_query[] = array(
+					'taxonomy' => 'post_format',
+					'field'    => 'slug',
+					'terms'    => array(),
+					'operator' => 'NOT EXISTS',
+				);
+				// Remove the standard format:
+				$formats = array_diff( $formats, array( 'standard' ) );
+			}
+
+			// Add any remaining formats to the tax query.
+			if ( ! empty( $formats ) ) {
+				// Add the post-format- prefix.
+				$terms = array_map(
+					function ( $format ) {
```

Comment:
```suggestion
					static function ( $format ) {
```

Micro-optimization: Closures/anon-functions in classes should use `static` if they're not accessing `$this`.